pkg/utils/apply/patch.go

```diff
@@ -169,3 +169,11 @@ func getOriginalConfiguration(obj runtime.Object) ([]byte, error) {
 	}
 	return []byte(original), nil
 }
+
+func isEmptyPatch(patch client.Patch) bool {
+	if patch == nil {
+		return true
+	}
+	data, _ := patch.Data(nil)
```

Comment:
why pass a nil into the patch.Data?

Reply:
Because that is the interface parameter. The generated patch already contain data.
Reference
1. https://github.com/kubernetes-sigs/controller-runtime/blob/cd0058ad295c268da1e7233e609a9a18dd60b5f6/pkg/client/patch.go#L48
2. https://github.com/kubevela/kubevela/blob/91638eba659ae2029d463b0f7aab26dfd7e18310/pkg/utils/apply/patch.go#L92